d3a_api_client/redis_aggregator.py

```diff
@@ -167,51 +168,32 @@ def batch_command(self, batch_command_dict, is_blocking=True):
 
     def _on_market_cycle(self, message):
         logging.info(f"A new market was created. Market information: {message}")
-
-        def executor_function():
-            try:
-                self.on_market_cycle(message)
-            except Exception as e:
-                root_logger.error(
-                    f"on_market_cycle raised exception: {str(e)}. \n Traceback: {str(traceback.format_exc())}")
-
-        self.executor.submit(executor_function)
+        function_name = "on_market_cycle"
+        function = lambda: self.on_market_cycle(message)
+        self.executor.submit(execute_function_util, function=function, function_name=function_name,
+                             root_logger=root_logger)
 
     def _on_tick(self, message):
         logging.info(f"Time has elapsed on the device. Progress info: {message}")
-
-        def executor_function():
-            try:
-                self.on_tick(message)
-            except Exception as e:
-                root_logger.error(
-                    f"on_tick raised exception: {e}. \n Traceback: {traceback.format_exc()}")
-
-        self.executor.submit(executor_function)
+        function_name = "on_tick"
+        function = lambda: self.on_tick(message)
+        self.executor.submit(execute_function_util, function=function, function_name=function_name,
+                             root_logger=root_logger)
 
     def _on_trade(self, message):
         logging.info(f"A trade took place on the device. Trade information: {message}")
+        function_name = "on_trade"
+        function = lambda: self.on_trade(message)
 
-        def executor_function():
-            try:
-                self.on_trade(message)
-            except Exception as e:
-                root_logger.error(
-                    f"_on_trade raised exception: {e}. \n Traceback: {traceback.format_exc()}")
-
-        self.executor.submit(executor_function)
+        self.executor.submit(execute_function_util, function=function, function_name=function_name,
+                             root_logger=root_logger)
```

Comment:
Is it really necessary to pass the root_logger here? How about running `root_logger = logging.getLogger()` in utils.py ?